SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialGDKDefaultLaunchConfigGenerator.h"
+
+#include "SpatialGDKEditorSettings.h"
+
+#include "Serialization/JsonWriter.h"
+#include "Misc/FileHelper.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialGDKDefaultLaunchConfigGenerator);
+
+#define LOCTEXT_NAMESPACE "SpatialGDKDefaultLaunchConfigGenerator"
+
+using namespace SpatialGDK;
+
+namespace
+{
+bool WriteFlagSection(TSharedRef<TJsonWriter<>> Writer, const FString& Key, const FString& Value)
+{
+	Writer->WriteObjectStart();
+		Writer->WriteValue(TEXT("name"), Key);
+		Writer->WriteValue(TEXT("value"), Value);
+	Writer->WriteObjectEnd();
+
+	return true;
+}
+
+bool WriteWorkerSection(TSharedRef< TJsonWriter<> > Writer, const FWorkerTypeLaunchSection& Worker)
+{
+	Writer->WriteObjectStart();
+		Writer->WriteValue(TEXT("worker_type"), *Worker.WorkerTypeName.ToString());
+		Writer->WriteArrayStart(TEXT("flags"));
+		for (const auto& Flag : Worker.Flags)
+		{
+			WriteFlagSection(Writer, Flag.Key, Flag.Value);
+		}
+		Writer->WriteArrayEnd();
+		Writer->WriteArrayStart(TEXT("permissions"));
+			Writer->WriteObjectStart();
+			if (Worker.WorkerPermissions.bAllPermissions)
+			{
+				Writer->WriteObjectStart(TEXT("all"));
+				Writer->WriteObjectEnd();
+			}
+			else
+			{
+				Writer->WriteObjectStart(TEXT("entity_creation"));
+					Writer->WriteValue(TEXT("allow"), Worker.WorkerPermissions.bAllowEntityCreation);
+				Writer->WriteObjectEnd();
+				Writer->WriteObjectStart(TEXT("entity_deletion"));
+					Writer->WriteValue(TEXT("allow"), Worker.WorkerPermissions.bAllowEntityDeletion);
+				Writer->WriteObjectEnd();
+				Writer->WriteObjectStart(TEXT("entity_query"));
+					Writer->WriteValue(TEXT("allow"), Worker.WorkerPermissions.bAllowEntityQuery);
+					Writer->WriteArrayStart("components");
+					for (const FString& Component : Worker.WorkerPermissions.Components)
+					{
+						Writer->WriteValue(Component);
+					}
+					Writer->WriteArrayEnd();
+				Writer->WriteObjectEnd();
+			}
+			Writer->WriteObjectEnd();
+		Writer->WriteArrayEnd();
+		if (Worker.MaxConnectionCapacityLimit > 0)
+		{
+			Writer->WriteObjectStart(TEXT("connection_capacity_limit"));
+				Writer->WriteValue(TEXT("max_capacity"), Worker.MaxConnectionCapacityLimit);
+			Writer->WriteObjectEnd();
+		}
+		if (Worker.bLoginRateLimitEnabled)
+		{
+			Writer->WriteObjectStart(TEXT("login_rate_limit"));
+				Writer->WriteValue(TEXT("duration"), Worker.LoginRateLimit.Duration);
+				Writer->WriteValue(TEXT("requests_per_duration"), Worker.LoginRateLimit.RequestsPerDuration);
+			Writer->WriteObjectEnd();
+		}
+	Writer->WriteObjectEnd();
+
+	return true;
+}
+
+bool WriteLoadbalancingSection(TSharedRef< TJsonWriter<> > Writer, const FName& WorkerType, const int32 Columns, const int32 Rows, const bool ManualWorkerConnectionOnly)
+{
+	Writer->WriteObjectStart();
+	Writer->WriteValue(TEXT("layer"), *WorkerType.ToString());
```

Comment:
Nit: Should be indented after WriteObjectStart to match the rest of the class

Reply:
Updated.